packages/core/src/utils.ts

```diff
@@ -115,11 +115,7 @@ export class Utils {
     signal: AbortSignal
   ) {
     if (typeof cid === 'string') cid = CID.parse(cid.replace('ipfs://', ''))
-    let format = await ipfsApi.codecs.getCodec(cid.code).then((f) => f.name)
-    // The try catch behaviour of the js-ipfs-http-client was causing timing issues (socket hangup)
-    // https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-http-client/src/block/put.js#L34
-    if (format === 'dag-cbor') format = 'cbor'
-    else if (format === 'dag-pb') format = 'protobuf'
+    const format = await ipfsApi.codecs.getCodec(cid.code).then((f) => f.name)
```

Comment:
Why are we removing the format handling? It seems like if we know the request is going to fail we should avoid it altogether if possible

Reply:
If we don't have any issues with it, it's probably safer to keep using the abstractions that ipfs provides in case they choose to change something under the hood?

Reply:
It's because it fails with `js-ipfs` with these changes :(

To reliably not make requests to either `js-ipfs` or `go-ipfs` that are unsuitable would require detecting the type of IPFS node through the `id` calls. That is an alternative though - make the right call to the right node.

We can do that since we make constant `id` calls to IPFS during our `healthcheck`, and `id` returns the node type, but it just seems a little hacky.